micro_ros_agent/src/agent/Agent.cpp

```diff
@@ -299,10 +334,17 @@ auto it = graph_manager_map_.find(domain_id);
     if (it != graph_manager_map_.end()) {
         return it->second;
     }else{
+        auto graph_manager = std::make_shared<graph_manager::GraphManager>(domain_id);
+        // Set namespace prefix if one was specified
+        if (!namespace_prefix_.empty())
+        {
+            graph_manager->set_namespace_prefix(namespace_prefix_);
+        }
+        
         return graph_manager_map_.insert(
             std::make_pair(
                 domain_id,
-                std::make_shared<graph_manager::GraphManager>(domain_id)
+                graph_manager
```

Comment:
Pass the namespace prefix through the constructor
```suggestion
                std::make_shared<graph_manager::GraphManager>(domain_id, namespace_prefix_)
```